bot.rb

```diff
@@ -26,6 +27,12 @@ class Bot
 
     attr_reader :logger
 
+    def self.should_notify_slack_message_not_that_important_though?
+      return true unless Date.today.saturday? || Date.today.sunday?
+      logger.info("Not notifying anyone because it's the weekend and people should enjoy their time off")
```

Comment:
`logger` is undefined here. Is `def self.name` defining a class method instead of instance method?

Reply:
```
diff --git a/bot.rb b/bot.rb
index 6f05a02..d1da722 100644
--- a/bot.rb
+++ b/bot.rb
@@ -27,13 +27,16 @@ module Fastlane
 
     attr_reader :logger
 
+    @@logger = nil
+
     def self.should_notify_slack_message_not_that_important_though?
       return true unless Date.today.saturday? || Date.today.sunday?
-      logger.info("Not notifying anyone because it's the weekend and people should enjoy their time off")
+      @@logger.info("Not notifying anyone because it's the weekend and people should enjoy their time off")
       return false
     end
 
     def initialize(logger)
+      @@logger = logger
       @logger = logger
     end
```

This seems to fix it.

Reply:
Ah, good point. Yeah, `self.name` is defining a class method